server/src/main/java/org/elasticsearch/cluster/metadata/SingleNodeShutdownMetadata.java

```diff
@@ -295,18 +331,35 @@ public Builder setAllocationDelay(TimeValue allocationDelay) {
             return this;
         }
 
+        /**
+         * @param targetNodeName The name of the node which should be used to replcae this one. Only valid if the shutdown type is REPLACE.
+         * @return This builder.
+         */
+        public Builder setTargetNodeName(String targetNodeName) {
+            this.targetNodeName = targetNodeName;
+            return this;
+        }
+
         public SingleNodeShutdownMetadata build() {
             if (startedAtMillis == -1) {
                 throw new IllegalArgumentException("start timestamp must be set");
             }
 
+            if (targetNodeName != null && type != Type.REPLACE) {
```

Comment:
I think doing this validate in the constructor instead would be better, since we do the validation for the shard delay there also